### ISSUE TITLE
Examine variant Seirawan board

### ISSUE THREAD

**Issue by ddugovic** [ddugovic/capablanca#3]
Thanks to @HGMuller 's helpful [comment](http://github.com/ddugovic/Stockfish/issues/147#issuecomment-264970124) I think I have [fixed](http://github.com/ddugovic/capablanca/commit/e904f87cc05823611631b5c164062c79ca717428) the `moves` command to output `seirawan/0` as the board type.

However, I assume the server is somehow sending wrong information, causing WinBoard to incorrectly display the start position after `ex b seirawan 0` (or `ex b wild seirawan`).

**Comment by HGMuller** [comment]
Op 12/6/2016 om 1:00 AM schreef Daniel Dugovic:
> However, I assume the server is somehow sending wrong information, 
> causing WinBoard to incorrectly display the start position after |ex b 
> seirawan 0| (or |ex b wild seirawan|). 
This is a WinBoard bug, you are probably using WB 4.8.0. Unfortuately I 
had broken variant recogition in ICS mode there, by making the matching 
of the variant name more strict. (It used to be very sloppy, just 
strstr(), which was causing problems with engine-defined variants like 
'minishogi', which would be recognized as 'shogi'.) I had not realized 
that in ICS mode the entire message "(Un)Rated XXX match" is fed to 
WinBoard's NameToVariant() routine. Later versions have this fixed, by 
going back to the old sloppy recognition in ICS mode.

The very latest WinBoard in the WB-AA package seems to have developed 
another bug, however. It correctly switches to a board with holdings, 
but the holdings do not hold the expected pieces, and are in fact 
accompanied by ridiculous counts. (It seems the counters are not 
properly zeroed, as they contain the code for 'EmptySquare'.) I will 
check this out, and keep you informed.

**Comment by HGMuller** [comment]
OK, I solved it. Turns out this bug must have been there from the 
beginning, it is just that no one ever tried to create an examined game 
for Seirawan Chess.

The problem is that the ICS first sends board, and then optionally 
holdings, without any variant indication. WinBoard then guesses the 
variant through a heuristic. When the ICS sends holdings, it was 
guessing Crazyhouse / Shogi / Great Shatranj based on the board width 
being 8 / 9 / 10. So in the Seirawan case it then switched to 
Crazyhouse, and started to interpret the ICS holdings command. This 
mentions the pieces H and E, however, which are unknown in Crazyhouse. 
In general, because the mapping pieceID -> pieceType -> holdingsSquare 
is variant dependent, using the Crazyhouse mapping messed up the 
holdings completely.

The problem is fixed by beefing up the variant-recognition heuristic, to 
not always assume Crazyhouse, but also look at the pieces that occur in 
the holdings, and interpret the presence of H or E there as an 
indication it must be a Seirawan Chess game.

Unfortunately the server for my webmail account seems to be 
malfunctioning at the moment, which prevents me from uploading the fixed 
WinBoard binary. I will put it in the WinBoard-AA package as soon as I 
can send e-mails again.
